variants/particle_xenon/variant.h

```diff
@@ -182,8 +182,24 @@ static const uint8_t SCK1  = PIN_SPI1_SCK;
 
 // On-board QSPI Flash
 #define EXTERNAL_FLASH_DEVICES   GD25Q16C
+
+#define USB_MSC_BLOCK_SIZE    512
+#define USB_MSC_BLOCK_COUNT   ((2*1024*1024) / USB_MSC_BLOCK_SIZE)
+
 #define EXTERNAL_FLASH_USE_QSPI
 
+static void switch_antenna(bool useExternal);
+
+// led pwm
```

Comment:
these functions header and implementation shouldn't be included by variant. These should be called by user sketches using PWM module or analogWrite(). I did remove these recently.

Reply:
Is there a place you'd recommend documenting that for a user? It's not obvious how to to switch antennas otherwise.

Reply:
@hathach analogWrite doesn't appear to do anything. Is it confirmed to work on other nRF52840 variants?

Reply:
switching antennas should be on Particle documentation. You could define the PIN_ANTENNA_N and example sketch to do that. variant should be about PIN definition and its on-board usage.

analogWrite work with 52840, check out this example here https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/libraries/Bluefruit52Lib/examples/Hardware/Fading/Fading.ino .

Reply:
That code you linked to does not work on the Particle Xenon. The original reason I added those led helper functions was to solve this very issue. For some reason, the existing analogWrite functions don't work on the Xenon.

Reply:
please find out why it doesn't work. Either way, these helpers still shouldn't be included in variant.

Reply:
Ok, I found out what was going on. It looks like the analogWrite and hardware pwm functions were using the `g_ADigitalPinMap` to figure out what pin to use (as opposed to the direct pin mappings in variant.h). I added all of the pins to `g_ADigitalPinMap` and now analogWrite works. I've removed the extraneous code from variant.cpp